src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -547,6 +547,7 @@ class DatatypeSegment(BaseSegment):
             Bracketed(Ref("NumericLiteralSegment"), optional=True),
             Sequence(OneOf("WITH", "WITHOUT"), "TIME", "ZONE", optional=True),
         ),
+        Sequence("DOUBLE", Sequence("PRECISION", optional=True),),
```

Comment:
Obviously I don't know all of the innards of this parser, and I am not an expert in ANSI SQL, but I did check to see if the "PRECISION" keyword is optional in Postgres, and it is not:

```
$ echo 'CREATE TABLE test ( angle double );' | psql -d test
ERROR:  type "double" does not exist
LINE 1: CREATE TABLE test ( angle double );
```

Reply:
Thanks! Updated.